### ISSUE TITLE
Using NoxPlayer with Android 12 and Kitsune wont install

### ISSUE THREAD

**Issue by inaciopestana** [HuskyDG/magisk-files#86]
I'm using NoxPlayer with Android 12 and Kitsune wont install with version 26.4+ (both stable and canary)
However, using the old 26.1 (Magisk Delta at the time) works. When I try to update inside the Kitsune manager to 26.4, i lost the magisk installation.

**Comment by VisionR1** [comment]
Check here maybe help you
https://github.com/HuskyDG/magisk-files/issues/8

**Comment by inaciopestana** [comment]
I read your linked thread and I did the following:

1) Installed Kitsune
2) Installed Kitsune directly on /system
3) Mounted system as RW
4) Created the folder sbin under /system/etc/init
5) Copied the magisk.rc that you linked in your post to that folder and renamed it to bootanim.rc
6) Reboot

It did not work. Under Kitsune, Magisk stills read as:
Installed: N/A

Did I do something wrong?

**Comment by VisionR1** [comment]
> I read your linked thread and I did the following:
> 
> 1. Installed Kitsune
> 2. Installed Kitsune directly on /system
> 3. Mounted system as RW
> 4. Created the folder sbin under /system/etc/init
> 5. Copied the magisk.rc that you linked in your post to that folder and renamed it to bootanim.rc
> 6. Reboot
> 
> It did not work. Under Kitsune, Magisk stills read as: Installed: N/A
> 
> Did I do something wrong?

Create the sbin folder under the main root /

**Comment by inaciopestana** [comment]
Thanks! That did the trick.

So, if anyone have the same problem, here it goes:

1. Installed Kitsune and granted root permissions
2. Installed Kitsune directly on /system
3. Mounted system as RW (mount -o rw,remount /system)
4. Created the folder sbin under root directory (/)
5. Copied [https://github.com/HuskyDG/magisk-files/files/13464588/bootanim.zip](this file) to /system/etc/init, replacing the original bootanim.rc that was already there.
6. Reboot

**Comment by VisionR1** [comment]
> Thanks! That did the trick.
> 
> So, if anyone have the same problem, here it goes:
> 
> 1. Installed Kitsune and granted root permissions
> 2. Installed Kitsune directly on /system
> 3. Mounted system as RW (mount -o rw,remount /system)
> 4. Created the folder sbin under root directory (/)
> 5. Copied [https://github.com/HuskyDG/magisk-files/files/13464588/bootanim.zip](url) to /system/etc/init, replacing the original bootanim.rc that was already there.
> 6. Reboot

That good, i think with latest Kitsune the bug with init file has been fixed

**Comment by inaciopestana** [comment]
I tried the latest and still doesnt work. But with your workaround, it did work!

**Comment by VisionR1** [comment]
> I tried the latest and still doesnt work. But with your workaround, it did work!

That's good to know

**Comment by inaciopestana** [comment]
When I ran it for the first time, the emulator restarted and, when Kitsune opened, Magisk appeared as installed, but asked to "repair the installation files" through a restart. Then, on the second restart, the emulator no longer booted. 
I tested on more than one clean image and got the same result.

**Comment by VisionR1** [comment]
> Please test this [app-debug.zip](https://github.com/HuskyDG/magisk-files/files/13990425/app-debug.zip)

I test this with latest Nox Android 12, the same still not created auto the sbin folder, and now when i create the sbin folder manually, and install Kitsune after reboot the Nox stuck at 99%, in previous version beside the sbin folder bug all other work great

**Comment by VisionR1** [comment]
> This is NoxPlayer bugs :))

Oh understand, so all working good, we can create the sbin folder manually

**Comment by inaciopestana** [comment]
Can't the creation of the sbin folder be part of the install script? I don't really understand the programing language, just wondering...

**Comment by VisionR1** [comment]
> We will fix it by using `/debug_ramdisk`

That's good. Keep the good work

**Comment by rahaaatul** [comment]
Still not fixed? 

@VisionR1 do you know why NOX keeps losing root after every reboot on android 12?

**Comment by VisionR1** [comment]
> Still not fixed?
> 
> @VisionR1 do you know why NOX keeps losing root after every reboot on android 12?

Mine works good Nox 7.0.5.9 Android 12, latest canary Kitsune Mask, you have still tick it the Root in nox settings ?

**Comment by rahaaatul** [comment]
> > Still not fixed?
> > @VisionR1 do you know why NOX keeps losing root after every reboot on android 12?
> 
> Mine works good Nox 7.0.5.9 Android 12, latest canary Kitsune Mask, you have still tick it the Root in nox settings ?

yes, but everytime i restart the root goes away (not kitsune mask).

but i put up with it, and installed kitsune stable successfully.

**Comment by VisionR1** [comment]
> > > Still not fixed?
> > > @VisionR1 do you know why NOX keeps losing root after every reboot on android 12?
> > 
> > Mine works good Nox 7.0.5.9 Android 12, latest canary Kitsune Mask, you have still tick it the Root in nox settings ?
> 
> yes, but everytime i restart the root goes away (not kitsune mask).
> 
> but i put up with it, and installed kitsune stable successfully.

Yeah, the old Magisk Delta, Nox have some problems, but if you create the sbin folder in the root before install the Delta, is works after, you can search here for the Magisk Delta with the Nox and other emulator, I have in past open issue and success fixed

**Comment by rahaaatul** [comment]
> > > > Still not fixed?
> > > > @VisionR1 do you know why NOX keeps losing root after every reboot on android 12?
> > > 
> > > 
> > > Mine works good Nox 7.0.5.9 Android 12, latest canary Kitsune Mask, you have still tick it the Root in nox settings ?
> > 
> > 
> > yes, but everytime i restart the root goes away (not kitsune mask).
> > but i put up with it, and installed kitsune stable successfully.
> 
> Yeah, the old Magisk Delta, Nox have some problems, but if you create the sbin folder in the root before install the Delta, is works after, you can search here for the Magisk Delta with the Nox and other emulator, I have in past open issue and success fixed

Yeah, thanks to that. I only created `sbin` folder and nothing else. It worked, and I didn't have copy any file in it. Although, I tried to, but the link in [here](https://github.com/HuskyDG/magisk-files/issues/86#issuecomment-1897229455) is misleading, couldn't find the file.

**Comment by VisionR1** [comment]
> > > > > Still not fixed?
> > > > > @VisionR1 do you know why NOX keeps losing root after every reboot on android 12?
> > > > 
> > > > 
> > > > Mine works good Nox 7.0.5.9 Android 12, latest canary Kitsune Mask, you have still tick it the Root in nox settings ?
> > > 
> > > 
> > > yes, but everytime i restart the root goes away (not kitsune mask).
> > > but i put up with it, and installed kitsune stable successfully.
> > 
> > 
> > Yeah, the old Magisk Delta, Nox have some problems, but if you create the sbin folder in the root before install the Delta, is works after, you can search here for the Magisk Delta with the Nox and other emulator, I have in past open issue and success fixed
> 
> Yeah, thanks to that. I only created `sbin` folder and nothing else. It worked, and I didn't have copy any file in it. Although, I tried to, but the link in [here](https://github.com/HuskyDG/magisk-files/issues/86#issuecomment-1897204449) is misleading, couldn't find the file.

That is good, yeah mine is this if you want the file is here
Check here maybe help you
https://github.com/HuskyDG/magisk-files/issues/8

**Comment by ToraxOutlaw** [comment]
I have manually created the sbin folder, installed Magisk Delta with Direct Install (modify /system directly), closed the reopened NoxPlayer and replaced the original bootanim.rc with the the suggested one, closed then reopened NoxPlayer. But after doing all that Magisk Delta still states that it is not installed.

What am I doing wrong?

**Comment by VisionR1** [comment]
> I have manually created the sbin folder, installed Magisk Delta with Direct Install (modify /system directly), closed the reopened NoxPlayer and replaced the original bootanim.rc with the the suggested one, closed then reopened NoxPlayer. But after doing all that Magisk Delta still states that it is not installed.
> 
> I am using NoxPlayer version 7.0.6.1001 and Android 12.
> 
> What am I doing wrong?

I think you the latest, no need the bootanim.rc, you have the root toggle?

**Comment by ToraxOutlaw** [comment]
Yes, I enabled the root toggle before I did anything.

**Comment by VisionR1** [comment]
> Yes, I enabled the root toggle before I did anything.

I mean after keep it enable.

**Comment by ToraxOutlaw** [comment]
Yes, I have always left i enabled.
![image](https://github.com/user-attachments/assets/d36c736b-cbde-4f28-a2b2-126de936443b)
![image](https://github.com/user-attachments/assets/404ad317-1e81-4fad-91fa-d4a04346e4ca)
![image](https://github.com/user-attachments/assets/b9070c98-2cdb-49fb-af46-446cb94da189)
![image](https://github.com/user-attachments/assets/e991e67e-e0ed-4ffa-80de-ceac0143b035)

**Comment by VisionR1** [comment]
> Yes, I have always left i enabled. ![image](https://private-user-images.githubusercontent.com/74446296/367512648-d36c736b-cbde-4f28-a2b2-126de936443b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjYzMTU0MjIsIm5iZiI6MTcyNjMxNTEyMiwicGF0aCI6Ii83NDQ0NjI5Ni8zNjc1MTI2NDgtZDM2YzczNmItY2JkZS00ZjI4LWEyYjItMTI2ZGU5MzY0NDNiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTE0VDExNTg0MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTUyMWM5NzMyZDNhZDU0YzZjMTY5ZTg5ZGE3MWMwN2FmNTRiZmYyZDk2MGE5ZDRmMzM1YmQ0NDg0YTExNmQ1NmEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.kcBvbcooAogeWKHXE7O17vV4kiPMwXjBxV1FOQS1BXw) ![image](https://private-user-images.githubusercontent.com/74446296/367512684-404ad317-1e81-4fad-91fa-d4a04346e4ca.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjYzMTU0MjIsIm5iZiI6MTcyNjMxNTEyMiwicGF0aCI6Ii83NDQ0NjI5Ni8zNjc1MTI2ODQtNDA0YWQzMTctMWU4MS00ZmFkLTkxZmEtZDRhMDQzNDZlNGNhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTE0VDExNTg0MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWRiNDgwYTYxM2YyOGZiZThlNTJkMzM2ZmFjYjk2ZjM0OGE5NTg1MGYxYWQyMjcwOWQ0MDNhZGJiZjhkNGIwMDQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.NmIvZFaPAXUYktvXuxKv6EwzNvL2uyrTZDrgGFjGoqU) ![image](https://private-user-images.githubusercontent.com/74446296/367512721-b9070c98-2cdb-49fb-af46-446cb94da189.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjYzMTU0MjIsIm5iZiI6MTcyNjMxNTEyMiwicGF0aCI6Ii83NDQ0NjI5Ni8zNjc1MTI3MjEtYjkwNzBjOTgtMmNkYi00OWZiLWFmNDYtNDQ2Y2I5NGRhMTg5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTE0VDExNTg0MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWIwMmJiMTNhOWE0MWU3NzllMmI0NDU0NzVlNTdkOTRkYTAwOTM3YzhhNGVkZjBhMGEwMzg5MWRmNmFlMGM4NWUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0._gpy5Utm4qtf_yGuAzInR_q9EytNOE7tIhjew2S2VOk) ![image](https://private-user-images.githubusercontent.com/74446296/367512784-e991e67e-e0ed-4ffa-80de-ceac0143b035.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjYzMTU0MjIsIm5iZiI6MTcyNjMxNTEyMiwicGF0aCI6Ii83NDQ0NjI5Ni8zNjc1MTI3ODQtZTk5MWU2N2UtZTBlZC00ZmZhLTgwZGUtY2VhYzAxNDNiMDM1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTE0VDExNTg0MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTdiZmE1ZTdlODI0ZjI0N2IxZTg0MGEwYWI3MjQ0YWNkOTg1MjYzNWU1NDYyNzYwMDc1ZjA0Y2JmZTZkNDkxZGEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.WuL0JZxKyh6pWxP7fFou2nyXZvZIh5rbYcmuAaHaGXM)

You have create the sbin folder in the wrong path.
The right is /sbin

**Comment by ToraxOutlaw** [comment]
Ahhhh. That would explain it. Give me a second and I'll let you know the outcome.
